xbmc/pvr/PVRGUIActions.cpp

```diff
@@ -700,6 +702,50 @@ namespace PVR
                                             CVariant{timer->Title()});
   }
 
+  class AsyncRecordingAction : private IRunnable
+  {
+  public:
+    bool Execute(const CFileItemPtr &item = CFileItemPtr());
```

Comment:
The performance impact is allocation of a pointer and atomic increment so it's not really a worry. 

If you prefer const ref I suggest passing the CFileItem as const ref instead of the shared_ptr. It avoids the extra allocation and looks saner to the reader. 

Anyway, I'm not in the habit of blocking pull requests so this is merely advice, you're free to pick whatever method you prefer as all issues are as you said mostly theoretical.